lib/jekyll/plugin_manager.rb

```diff
@@ -57,6 +55,18 @@ def self.require_from_bundler
       end
     end
 
+    # Build an array of enabled plugin names.
+    #
+    # Returns an array of strings, each string being the name of a plugin
+    #   that is specified (and whitelisted in case the site is in safe mode).
+    def enabled_plugins
+      @enabled_plugins ||= if site.safe
+                             site.gems & whitelist
```

Comment:
I assumed a plugin is only enabled if it is both listed in the `site.gems` (`plugins` config key) and in the `whitelist`? Isn't the whitelist not only a filter for this list for safe plugins?
```ruby
a = ["foo", "bar"]
b = ["bar", "baz"]
a & b  # => ["bar"]
a && b # => ["bar", "baz"]